packages/tiled/lib/src/tiled_map.dart

```diff
@@ -219,6 +219,36 @@ class TiledMap {
     return imageSet.toList();
   }
 
+  List<TiledImage> collectImagesInLayer(Layer layer) {
+    if (layer is ImageLayer) {
+      return [layer.image];
+    } else if (layer is Group) {
+      return layer.layers.map(collectImagesInLayer).flattened.toList();
```

Comment:
nit: I think you can use `expand` instead of `map + flattened`

Reply:
Haha I love how everyone had their own suggestion, this seems like the best one though. :)

Reply:
I realized that I had always been using `map + expand` rather foolishly 😂